File: WoofIOS/SwiftPackages/Presentation/Account/Sources/PresentationAccount/UI/AccountAdapter.swift
import SwiftUI

public struct AccountAdapter: View {
    @StateObject private var viewModel: AccountViewModel

    public init() {
        _viewModel = .init()
    }

    public var body: some View {
        AccountScreen(state: viewModel.state)
            .onAppear {
                viewModel.onViewAppear()
            }
    }
}

File: WoofAndroid/presentation/account/src/main/kotlin/com/perrystreet/woof/presentation/account/ui/AccountAdapter.kt
package com.perrystreet.woof.presentation.account.ui

import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import com.perrystreet.woof.presentation.account.viewmodel.AccountViewModel
import org.koin.androidx.compose.koinViewModel

@Composable
fun AccountAdapter(viewModel: AccountViewModel = koinViewModel()) {
    val state by viewModel.state.subscribeAsState()

    LaunchedEffect(Unit) {
        viewModel.onViewAppear()
    }

    AccountScreen(state = state)
}
